src/snowflake/connector/cursor.py

```diff
@@ -748,11 +749,18 @@ def execute(
             logger.warning("execute: no query is given to execute")
             return None
 
-        if _statement_params is None:
-            _statement_params = dict()
-
+        extra_statement_params = dict()
```

Comment:
NIT:
How about we put
```
_statement_params = (_statement_params or dict()).copy()
```
before the if case?

Reply:
I think that will make us always copy the statement parameters, which I'm not sure if it is a good idea. Though what I did might also well be an overkill 🤔

Reply:
I see, what about:
```
_statement_params = _statement_params or dict()
if num_statements:
    _statements_params = {**_statements_params, 'MULTI_STATEMENT_COUNT':num_statements}
```
?

Reply:
Hmmm but `extra_statement_params` is probably more extensible in case we want to add more parameters in the future. 

Maybe we could change the extra param check to:

```
        extra_statement_params = dict()
        if num_statements:
            extra_statement_params["MULTI_STATEMENT_COUNT"] = num_statements
        if extra_statement_params:
            _statement_params = {**(_statement_params or dict()), **extra_statement_params)}
```

Reply:
As discussed offline with @sfc-gh-stan, we will add a comment about `extra_statement_params` as a TODO when we need to add another statement. For now, we will add fewer lines of code.